contracts/PushComm/PushCommV3.sol

```diff
@@ -554,111 +563,140 @@ contract PushCommV3 is Initializable, PushCommStorageV2, IPushCommV3, PausableUp
-        bytes memory requestPayload = abi.encode(arbitraryReqPayload, msg.sender, CrossChainRequestTypes.RequestType.ArbitraryReq);
-        createCrossChainRequest(requestPayload, _amount, _gasLimit);
+        ADD_CHANNEL_MIN_FEES = _minChannelCreationFee;
+        FEE_AMOUNT = _feeAmount;
+        PROTOCOL_POOL_FEE = _protocolPoolFee;
     }
 
+    /**
+     * @notice Quotes the cost of bridging tokens to the recipient chain
+     * @dev Calls the Wormhole Transceiver to get the delivery price
+     * @return cost The cost of bridging tokens
+     */
     function quoteTokenBridgingCost() public view returns (uint256 cost) {
         TransceiverStructs.TransceiverInstruction memory transceiverInstruction =
             TransceiverStructs.TransceiverInstruction({ index: 0, payload: abi.encodePacked(false) });
         cost = WORMHOLE_TRANSCEIVER.quoteDeliveryPrice(WORMHOLE_RECIPIENT_CHAIN, transceiverInstruction);
     }
 
+    /**
+     * @notice Quotes the cost of relaying a message to the target chain with the specified gas limit
+     * @dev Calls the Wormhole Relayer to get the EVM delivery price
+     * @param targetChain The chain to which the message is being relayed
+     * @param gasLimit The gas limit for the message relay
+     * @return cost The cost of relaying the message
+     */
     function quoteMsgRelayCost(uint16 targetChain, uint256 gasLimit) public view returns (uint256 cost) {
         (cost,) = WORMHOLE_RELAYER.quoteEVMDeliveryPrice(targetChain, 0, gasLimit);
     }
 
-    function createCrossChainRequest(bytes memory _requestPayload, uint256 _amount, uint256 _gasLimit) public payable {
-        bytes32 recipient = BaseHelper.addressToBytes32(EPNSCoreAddress);
-        bytes32 sender = BaseHelper.addressToBytes32(msg.sender);
+    /**
+     * @notice Creates a cross-chain request based on the specified function type and payload
+     * @dev Implements restrictions and calls the internal function to create the cross-chain request
+     * @param functionType The type of cross-chain function to execute
+     * @param payload The payload data for the cross-chain request
+     * @param amount The amount of tokens to be transferred
+     * @param gasLimit The gas limit for the cross-chain request
+     */
+    function createCrossChainRequest(
+        CrossChainRequestTypes.CrossChainFunction functionType,
+        bytes calldata payload,
+        uint256 amount,
+        uint256 gasLimit
+    )
+        external
+        payable
+        whenNotPaused
+    {
+        // Implement restrictions based on functionType
+        if (functionType == CrossChainRequestTypes.CrossChainFunction.AddChannel) {
+            if (amount < ADD_CHANNEL_MIN_FEES) {
+                revert Errors.InvalidArg_LessThanExpected(ADD_CHANNEL_MIN_FEES, amount);
+            }
+        } else if (functionType == CrossChainRequestTypes.CrossChainFunction.IncentivizedChat) {
+            if (amount == 0 && amount < FEE_AMOUNT) {
```

Comment:
We can put only one check in this line. 

if the amount is zero then it will be less than FEE_AMOUNT anyway.  
I think, this line would suffice:
`if (amount < FEE_AMOUNT)`